designs/disruption-termination-grace-period.md

```diff
@@ -0,0 +1,123 @@
+# Disruption Termination Grace Period
+
+## Motivation
+Users are requesting the ability to control how long karpenter will wait for the deletion of an individual node to complete before forcefully terminating the node regardless of pod status. [#743](https://github.com/kubernetes-sigs/karpenter/issues/743). This supports two primary use cases.
+* Cluster admins who want to ensure that nodes are cycled after a given period of time, regardless of user-defined disruption controls (such as PDBs or preStop hooks) that might prevent eviction of a pod beyond the configured limit. This could be to satisfy security requirements or for convenience.
+* Cluster admins who want to allow users to protect long-running jobs from being interrupted by node disruption, up to a configured limit.
+
+This design piggybacks off the existing disruption-controls design, adding in a new `terminationGracePeriodSeconds` field to the `v1beta/NodePool` `Disruption` spec. This field seems intuitively related to the other disruption controls that exist within the disruption block.
+
+## Proposed Spec
+
+```yaml
+apiVersion: karpenter.sh/v1beta1
+kind: NodePool
+metadata:
+  name: default
+spec: # This is not a complete NodePool Spec.
+  disruption:
+    consolidationPolicy: WhenUnderutilized || WhenEmpty
+    consolidateAfter: 10m || Never # metav1.Duration
+    expireAfter: 10m || Never # Equivalent to v1alpha5 TTLSecondsUntilExpired
+    terminationGracePeriodSeconds: 86400 || 24h || nil
```

Comment:
if we are specifying seconds we shouldn't allow 24h?

Reply:
Yeah, you're correct. I'm going to update it to support standard duration formats for convenience so we don't have to google how many seconds are in a day all the time.

Reply:
What about terminationGracePeriodDuration? Would make it easier to do the thing you're trying to do.

Reply:
> What about terminationGracePeriodDuration? Would make it easier to do the thing you're trying to do.
Happy with that or just terminationGracePeriod

Reply:
I think I slightly prefer `terminationGracePeriod` just for brevity, but I don't have very strong opinions.